### PR TITLE
Data timestamp version in response

### PR DESCRIPTION
**Pull request by gardster**
# Issue

[5115](https://github.com/Project-OSRM/osrm-backend/issues/5115)

### CODE REVIEW

**Comment by gardster** [comment]
Hi! Can anyone review my PR, please?
@TheMarex

**Comment by danpat** [comment]
@gardster Overall, I think this idea is useful, but before merging it, I think it needs some changes:

  1. Make the `data_version` field optional in the response - don't return a fixed `n/a` value if it's missing, instead exclude the field altogether.
  2. Add a command-line flag to `osrm-extract` called something like `--data-version X`, where X can be one of:
    - `none` - would not save the version, and will cause API response to have no `data_version` field
    - `osmosis_replication_timestamp` - will extract the Osmosis timestamp from the OSM file - I think it would be reasonable to make this the default
    - `custom:<string>` - allow the user to supply a string on the command-line that'll get saved

The reason for these changes - I don't think users should be required to expose their data version to users if they don't want to.  Here at Mapbox, we don't even guarantee that we're exposing a single version of OSM data - we validate changesets individually, so some changes may get delayed while others come through.  Until we figure out what `data_version` would mean in that case, we wouldn't want `n/a` in every API response - it's not useful information (this is why I suggest the `custom:<string>` option above).

There's also the added complication of traffic updates.  It's possible to run `osrm-extract` once, then `osrm-customize` or `osrm-contract` multiple times with updated traffic data.  In this case, `data_version` wouldn't reflect all the information you might want to know about the data.  I'm not sure whether we'd want a second field for that, or whether `osrm-contract/osrm-customize` should be able to modify `data_version` in some way....

**Comment by gardster** [comment]
Thank you for the comment.
I will implement this functionality.

**Comment by gardster** [comment]
PTAL

File: include/engine/api/route_api.hpp [hunk 1]
```diff
@@ -68,6 +68,11 @@ class RouteAPI : public BaseAPI
         response.values["waypoints"] = BaseAPI::MakeWaypoints(all_start_end_points);
         response.values["routes"] = std::move(jsRoutes);
         response.values["code"] = "Ok";
+        auto data_timestamp = facade.GetTimestamp();
+        if (!data_timestamp.empty())
+        {
+            response.values["data_version"] = facade.GetTimestamp();
```
Comment (danpat) [inline include/engine/api/route_api.hpp]: This could just be `response.values["data_version"] = data_timestamp;`, right?

**Comment by danpat** [comment]
@gardster If you could add a couple of tests, I think this is good to merge 👍

File: include/engine/api/route_api.hpp (hunk 1 above)
Comment (gardster) [reply include/engine/api/route_api.hpp]: Sure

**Comment by gardster** [comment]
PTAL

**Comment by danpat** [comment]
@gardster Merged with https://github.com/Project-OSRM/osrm-backend/commit/8b45ff7a186d158345304169768cd7d99acb8126 thanks for the feature!

**Pull request closed without merging**